core/api/src/servers/graphql-admin-api-server.ts

```diff
@@ -64,32 +74,62 @@ const setGqlAdminContext = async (
   )
 }
 
-const isAuthenticated = rule({ cache: "contextual" })(async (
+// Feature-specific authorization rules
+const requiresViewAccess = rule({ cache: "contextual" })(async (
   parent,
   args,
   ctx: GraphQLAdminContext,
 ) => {
-  return (
-    ctx.privilegedClientId !== null &&
-    ctx.privilegedClientId !== undefined &&
-    ctx.privilegedClientId !== ""
-  )
+  if (!ctx.userEmail) return false
+  return roleChecker.hasFeature(ctx.userEmail, AdminFeature.VIEW_ACCOUNTS)
+})
+
+const requiresModifyAccess = rule({ cache: "contextual" })(async (
+  parent,
+  args,
+  ctx: GraphQLAdminContext,
+) => {
+  if (!ctx.userEmail) return false
+  return roleChecker.hasFeature(ctx.userEmail, AdminFeature.MODIFY_ACCOUNTS)
 })
 
 export async function startApolloServerForAdminSchema() {
-  const authedQueryFields: { [key: string]: Rule } = {}
-  for (const key of Object.keys(adminQueryFields.authed)) {
-    authedQueryFields[key] = isAuthenticated
+  // View-only queries
+  const viewerQueryFields: { [key: string]: Rule } = {}
+  const viewerQueries = [
+    "accountDetailsByUserId",
+    "accountDetailsByPhone",
+    "accountDetailsByEmail",
+  ]
+
+  for (const key of viewerQueries) {
+    if (key in adminQueryFields.authed) {
+      viewerQueryFields[key] = requiresViewAccess
+    }
+  }
+
+  // Modification queries
+  const modifyQueryFields: { [key: string]: Rule } = {}
+  const modifyQueries = ["updateUserPhone", "updateUserEmail"]
+
+  for (const key of modifyQueries) {
+    if (key in adminQueryFields.authed) {
+      modifyQueryFields[key] = requiresModifyAccess
+    }
   }
```

Comment:
this strategy is prone to errors please create the proper groups in core/api/src/graphql/admin/mutations.ts and core/api/src/graphql/admin/queries.ts